servers/services/src/main/java/org/projectnessie/services/hash/ResolvedHash.java

```diff
@@ -36,29 +34,13 @@ public interface ResolvedHash extends WithHash<NamedRef> {
   /** The {@link NamedRef}; can be {@link DetachedRef}. */
   NamedRef getNamedRef();
 
-  /**
-   * The user-provided hash. Only present if a hash was actually provided and successfully parsed;
-   * will be empty if the hash was implicitly resolved to the ref's HEAD.
-   *
-   * <p>If this is empty, then {@link #getHead()} is guaranteed to be non-empty.
-   *
-   * <p>If the provided hash had relative specs, this hash will be the result of resolving those
-   * specs against its absolute part, or the ref's HEAD if it doesn't have an absolute part.
-   */
-  Optional<Hash> getProvidedHash();
-
   /**
    * The ref's HEAD, if available. Will always be empty for {@link DetachedRef}. Exposed mostly to
    * avoid re-fetching the HEAD many times.
-   *
-   * <p>If this is empty, then {@link #getProvidedHash()} is guaranteed to be non-empty.
    */
   Optional<Hash> getHead();
 
-  /**
-   * The effective resolved hash, never {@code null}. Will either be {@link #getProvidedHash()} if
-   * present, or {@link #getHead()} otherwise.
-   */
+  /** The effective resolved hash, never {@code null}. */
```

Comment:
nit: why not annotate as `@Nonnull`?

Reply:
The Immutables library has no support for `@Nonnull`: it encourages instead to consider everything as non-null and annotate the exceptions with `@Nullable` or use `Optional`. See http://immutables.github.io/immutable.html#nullable-attributes.